network/network.go

```diff
@@ -1331,3 +1372,59 @@ func (n *network) restartOnDisconnect() {
 		}
 	}
 }
+
+// HealthCheck returns information about several network layer health checks.
+// 1) Information about health check results
+// 2) An error if the health check reports unhealthy
+// Assumes [n.stateLock] is not held
+func (n *network) HealthCheck() (interface{}, error) {
+	details := map[string]interface{}{}
+	healthy := true
+
+	// Get some data with the state lock held
+	connectedTo := 0
+	n.stateLock.RLock()
+	for _, peer := range n.peers {
+		if peer != nil && peer.connected.GetValue() {
+			connectedTo++
+			if connectedTo > int(n.healthConfig.MinConnectedPeers) {
+				break
+			}
+		}
+	}
+	pendingSendBytes := n.pendingBytes
+	sendFailRate := n.sendFailRateCalculator.Read()
+	n.stateLock.RUnlock()
+
+	// Make sure we're connected to at least the minimum number of peers
+	isSufficientlyConnected := connectedTo >= int(n.healthConfig.MinConnectedPeers)
+	healthy = healthy && isSufficientlyConnected
```

Comment:
If healthy was defined here, then we could just originally set it to `isSufficientlyConnected`.